vision/pipeline.py

```diff
@@ -88,44 +93,92 @@ def run(self, prev_state):
             state = self.flight_communication.get_nowait()
         except Empty:
             state = prev_state
-
         ##
+
         bboxes = []
 
+        flags = {}
+
+        state = "module_detection"
+
         if state == "early_laps":  # navigation around the pylons
-            bboxes = self.obstacle_finder.find(color_image, depth_image)
+            flags = ObstacleDetectionFlags()
+
+            try:
+                bboxes = self.obstacle_finder.find(color_image, depth_image)
+                self.obstacle_tracker.update(bboxes)
+            except:
+                flags.obstacle_finder = False
 
-            obstacle_tracker.update(bboxes)
-            bboxes = obstacle_tracker.getPersistentObstacles()
+            try:
+                bboxes = self.obstacle_tracker.getPersistentObstacles()
+            except:
+                flags.obstacle_tracker = False
 
         elif state == "text_detection":  # approaching mast
-            bboxes = self.text_detector.detect_russian_word(color_image, depth_image)
+            flags = TextDetectionFlags()
+
+            try:
+                bboxes = self.text_detector.detect_russian_word(color_image, depth_image)
+            except:
+                flags.detect_russian_word = False
 
         elif state == "module_detection":  # locating module
-            self.module_location.set_img(color_image, depth_image)
+            flags = ModuleDetectionFlags()
+            try:
+                self.module_location.setImg(color_image, depth_image)
+            except:
+                flags.setImg = False
+
+            try:
+                inFrame = self.module_location.isInFrame()
+            except:
+                inFrame = False
+                flags.isInFrame = False
 
             # only do more calculation if module is in the image
-            if self.module_location.is_in_frame():
-                center = self.module_location.get_center()  # center of module in image
-                depth = get_module_depth(
-                    depth_image, center
-                )  # depth of center of module
-
-                region = region_of_interest(
-                    depth_image, depth, center
-                )  # depth image sliced on underestimate bounds
-                orientation = get_module_orientation(
-                    region, center
-                )  # x and y tilt of module
-
-                bounds = get_module_bounds(
-                    color_image, center, depth
-                )  # overestimate of bounds
-                roll = get_module_roll(
-                    color_image[
-                        bounds[0][1] : bounds[3][1], bounds[0][0] : bounds[2][0], :
-                    ]
-                )  # roll of module
+            if inFrame:
+                try:
+                    center = self.module_location.getCenter()  # center of module in image
+                except:
+                    flags.getCenter = False
+
+                try:
```

Comment:
Same thing with a lot of the module algorithms. These should flow mostly linearly except for where get_module_bounds doesn't depend on region/orientation.